Firmware/sound.h

```diff
@@ -13,6 +13,9 @@ typedef enum
 typedef enum
      {e_SOUND_CLASS_Echo,e_SOUND_CLASS_Prompt,e_SOUND_CLASS_Confirm,e_SOUND_CLASS_Warning,e_SOUND_CLASS_Alert} eSOUND_CLASS;
 
+// The minimum number of milliseconds of sound duration, to use the keep-alive method instead of the default delay
+// This is used to allow the printer to retain control in long-running sound output
+#define SOUND_KEEPALIVE_MIN_MS 1100
```

Comment:
Not really relevant considering the points above, but just a quick observation:
Instead of doing `if (delay < 1100) delay(); else delay_keep_alive();`, do `if (delay > 1000) delay_keep_alive(); else delay();`

Reply:
I didn't just choose 1100 ms to exclude one second long beeps, but I added 100 ms in case someone somewhere used a delay around 1 second, but not exactly.